src/property.ts

```diff
@@ -215,27 +215,22 @@ export class PropertiesManager {
   }
 
   set(propertiesToSet: Properties): void {
-    this.properties = {
-      ...Object.keys(propertiesToSet).map((propertyName) => [
-        propertyName,
-        get(propertyName),
-      ]),
-      ...this.properties,
-    };
     Object.entries(propertiesToSet).forEach(([propertyName, propertyValue]) => {
+      if (!((propertyName as KnownProperty) in this.properties)) {
+        this.properties[propertyName as KnownProperty] = get(propertyName);
+      }
       set(propertyName, propertyValue);
     });
   }
 
   setChoices(choicesToSet: { [choice: number]: number | string }): void {
-    this.set(
-      Object.fromEntries(
-        Object.entries(choicesToSet).map(([choiceNumber, choiceValue]) => [
-          `choiceAdventure${choiceNumber}` as NumericOrStringProperty,
-          choiceValue,
-        ])
-      )
-    );
+    Object.entries(choicesToSet).forEach(([choiceNumber, choiceValue]) => {
```

Comment:
Why not use this.set? DRY

Reply:
I originally did that, but my concern was that I wasn't sure the object I was constructing in the way I set it up would actually act as a Properties thing. I've changed it to use this.set, but make sure to give that section a careful look to make sure it's not just entirely wrong